bot/commands/FeedbackCommands.cs

```diff
@@ -1,90 +1,199 @@
 {
+    private readonly IConfig config = config;
     private readonly ILogger<FeedbackCommands> logger = logger;
+    private readonly IHttpClientFactory httpClientFactory = httpClientFactory;
 
     public Dictionary<string, string> Commands => new()
     {
         { "/feedback", "shows your rating and comments for the last response from the bot." },
         { "/rate up", "rates the last response from the bot as positive." },
         { "/rate down", "rates the last response from the bot as negative." },
-        { "/rate-comment ???", "allows you to give feedback comments on the last response from the bot."  },
+        { "/rate-comment <rating_comment>", "allows you to give feedback comments on the last response from the bot."  },
         { "/revoke", "to clear your rating and comments for the last response from the bot."  },
     };
 
     public async Task<bool> Try(ITurnContext<IMessageActivity> turnContext, CancellationToken cancellationToken = default)
     {
         var userId = turnContext.Activity.From.AadObjectId;
 
-        // look for actions
-        var jaction = turnContext.Activity.Value as JObject;
-        var action = jaction?.ToObject<UserAction>();
-        if (action is not null && !string.IsNullOrEmpty(action.ChatId))
+        if (turnContext.Activity.Value is JObject jaction && jaction.ToObject<UserAction>() is UserAction action && !string.IsNullOrEmpty(action.ActivityId))
         {
-            switch (action.Command)
-            {
-                case "/rate up":
-                    {
-                        await this.Rate(userId, action.ChatId, "up", turnContext, cancellationToken);
-                        break;
-                    }
-                case "/rate down":
-                    {
-                        await this.Rate(userId, action.ChatId, "down", turnContext, cancellationToken);
-                        break;
-                    }
-            }
+            // Check the result here, if it's true, return true. Otherwise, continue to the next step.
+            var result = await HandleActionAsync(userId, action, turnContext, cancellationToken);
+            if (result) { return true; }
         }
 
-        // look for commands
-        switch (turnContext.Activity.Text?.ToLower())
+        var text = turnContext.Activity.Text?.ToLower();
+        if (text != null)
+        {
+            return await HandleCommandAsync(userId, text, turnContext, cancellationToken);
+        }
+
+        return false;
+    }
+
+    private async Task<bool> HandleActionAsync(string userId, UserAction action, ITurnContext<IMessageActivity> turnContext, CancellationToken cancellationToken)
+    {
+        switch (action.Command)
+        {
+            case "/rate up":
+#pragma warning disable CS8604 // Possible null reference argument.
+                await RateAsync(userId, action.ActivityId, "up", turnContext, cancellationToken);
```

Comment:
I assume this is related to ActivityId. Maybe try ! at the end instead of disabling the warning. Alternatively, consider passing the activityId as the parameter.